src/controller/python/chip/yaml/runner.py

```diff
@@ -0,0 +1,387 @@
+                                                    self._request_object))
+        except chip.interaction_model.InteractionModelError as error:
+            return _ActionResult(status=_ActionStatus.ERROR, response=error)
+
+        # Commands with no response give a None response. In those cases we return a success
+        return _ActionResult(status=_ActionStatus.SUCCESS, response=resp)
+
+
+class ReadAttributeAction(BaseAction):
+    '''Single read attribute action to be executed.'''
+
+    def __init__(self, test_step, cluster: str, context: _ExecutionContext):
+        '''Converts 'test_step' to read attribute action that can execute with ChipDeviceCtrl.
+
+        Args:
+          'test_step': Step containing information required to run read attribute action.
+          'cluster': Name of cluster read attribute action is targeting.
+          'context': Contains test-wide common objects such as DataModelLookup instance.
+        Raises:
+          ParsingError: Raised if there is a benign error, and there is currently no
+            action to perform for this read attribute.
+          UnexpectedParsingError: Raised if there is an unexpected parsing error.
+        '''
+        super().__init__(test_step.label)
+        self._attribute_name = stringcase.pascalcase(test_step.attribute)
+        self._cluster = cluster
+        self._endpoint = test_step.endpoint
+        self._node_id = test_step.node_id
+        self._cluster_object = None
+        self._request_object = None
+
+        self._possibly_unsupported = bool(test_step.optional)
+
+        self._cluster_object = context.data_model_lookup.get_cluster(self._cluster)
+        if self._cluster_object is None:
+            raise UnexpectedParsingError(
+                f'ReadAttribute failed to find cluster object:{self._cluster}')
+
+        self._request_object = context.data_model_lookup.get_attribute(
+            self._cluster, self._attribute_name)
+        if self._request_object is None:
+            raise ParsingError(
+                f'ReadAttribute failed to find cluster:{self._cluster} '
+                f'Attribute:{self._attribute_name}')
+
+        if test_step.arguments:
+            raise UnexpectedParsingError(
+                f'ReadAttribute should not contain arguments. {self.label}')
+
+        if self._request_object.attribute_type is None:
+            raise UnexpectedParsingError(
+                f'ReadAttribute doesnt have valid attribute_type. {self.label}')
+
+    def run_action(self, dev_ctrl: ChipDeviceCtrl) -> _ActionResult:
+        try:
+            raw_resp = asyncio.run(dev_ctrl.ReadAttribute(self._node_id,
+                                                          [(self._endpoint, self._request_object)]))
+        except chip.interaction_model.InteractionModelError:
+            # TODO Should we be doing the same thing as InvokeAction on InteractionModelError?
```

Comment:
>Does it throw if there are errors with the command going out?

I have only seen timeout because node didn't exist and there was an exception for that

>Would it make sense to tuck the error into an action response?

My thought process was raising exception here for the mean time instead of returning an error here means that someone who is expecting a failure here could validate what we want to happen and easily see the comment with a  solution. But I am okay with going in either direction

Reply:
Done